myJoin.pl

```diff
@@ -52,22 +69,30 @@
 #Duplicate line names removed, keys can not be duplicated in hash.
 #increment any names repeated more than once, indicates they are duplicates.
 
+#subroutine to find duplicates within and between files
 sub line_hash1{
+    my %observed;
     my $input = shift;
     open(FILE, $input) or die $!;
     if ($header){
         $header_text = <FILE>;
         chomp $header_text;
-    }
-    while (<FILE>){
-        chomp;
-        $line1{$_} += 1;
-    }
+        }
+        while (<FILE>){
+            chomp;
+            #hash to remove duplicate names from individual file
+            $observed{$_}++;
+            next if $observed{$_} > 1;
+                    }
+                    #hash to find duplicate names between files
+                    foreach my $unq (sort keys %observed){
```

Comment:
This is unnecessary. If we make it past the `next if` statement in the above block, then it is the first time we've observed a line in this file. Inserting a line in that block to update `%line1` for the first observation of a line in a file would be more efficient.